python/src/iceberg/io/base.py

```diff
@@ -76,7 +76,15 @@ def close(self) -> None:
 
 
 class InputFile(ABC):
-    """A base class for InputFile implementations"""
+    """A base class for InputFile implementations
+
+    Args:
+        location(str): A URI or a path to a local file
+
+    Attributes:
+        location(str): The URI or path to a local file for an InputFile instance
+        exists(bool): Whether the file exists or not
+    """
```

Comment:
I just realized that we're missing a method for both `InputFile` and `OutputFile`: `close`! It's not needed to read or write, but it's really important!

Reply:
We have `close` as part of the `InputStream` and `OutputStream` protocol. I don't think we manage the file object's state in the `InputFile` and `OutputFile` classes, do we still need a `close` method in that case?

Reply:
I must have missed it. I didn't see it on the protocol, but that's where I think it should be.